iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeeders.java

```diff
@@ -0,0 +1,153 @@
+                notFoundBusbarSectionReport(reporter, bbsId);
+                if (throwException) {
+                    throw new PowsyblException(String.format("Busbar section %s not found.", bbsId));
+                }
+                return;
+            }
+            VoltageLevel voltageLevel = bbs.getTerminal().getVoltageLevel();
+            int connectableNode = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;
+            setNode(side, connectableNode, voltageLevel.getId());
+        }
+
+        Connectable<?> connectable = add();
+        if (connectable.getNetwork() != network) {
+            connectable.remove();
+            LOGGER.error("Network given in parameters and in connectableAdder are different. Connectable '{}' of type {} was added then removed",
+                    connectable.getId(), connectable.getType());
+            networkMismatchReport(reporter, connectable.getId(), connectable.getType());
+            if (throwException) {
+                throw new PowsyblException("Network given in parameters and in connectableAdder are different. Connectable was added then removed");
+            }
+            return;
+        }
+        String connectableId = connectable.getId();
+
+        boolean createConnectablePosition = false;
+        ConnectablePositionAdder<?> connectablePositionAdder = connectable.newExtension(ConnectablePositionAdder.class);
+        for (int side : sides) {
+            VoltageLevel voltageLevel = getVoltageLevel(side, connectable);
+            Set<Integer> takenFeederPositions = TopologyModificationUtils.getFeederPositions(voltageLevel);
+            int positionOrder = getPositionOrder(side);
+            if (!takenFeederPositions.isEmpty()) {
+                if (takenFeederPositions.contains(positionOrder)) {
+                    LOGGER.error("PositionOrder {} already taken.", positionOrder);
+                    positionOrderAlreadyTakenReport(reporter, positionOrder);
+                    if (throwException) {
+                        throw new PowsyblException(String.format("PositionOrder %d already taken.", positionOrder));
+                    }
+                    return;
+                }
+                getFeederAdder(side, connectablePositionAdder)
+                        .withDirection(getDirection(side))
+                        .withOrder(positionOrder)
+                        .withName(connectableId)
+                        .add();
+                createConnectablePosition = true;
+            } else {
+                LOGGER.warn("No extensions found on voltageLevel {}. The extension is not created.", voltageLevel.getId());
+                noConnectablePositionExtension(reporter, voltageLevel);
+            }
+            // create switches and a breaker linking the connectable to the busbar sections.
+            int node = getNode(side, connectable);
+            createTopology(side, network, voltageLevel, node, node + 1, connectable, reporter);
```

Comment:
here we suppose that the node used for the connectable has a spot available just after. We could be more robust by just doing `int forkNode = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1` inside `createTopology`.

Reply:
The node used is already `int connectableNode = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;` so the spot just after is always available, this is why we only added 1!

Reply:
Yes, now it's ok, but if one day you add another node between `setNode` and `getNode`, then it's not working anymore. That was what I meant by "more robust".

Reply:
Which makes me think about the special case of a branch with both side in the same voltage level! The fork nodes would then be `getNode(side, connectable) + 2`.

Reply:
Ah yes, you're right... Changing it

Reply:
you probably got it, but it seems that for that special case, the current code in the setNode loop results in the same node for both sides.

Reply:
Indeed... But I don't think this case ever happens? Theoretically it is allowed but I don't think it does.

A quick hack can be to use side even if it create holes:
`int connectableNode = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + (side == 0 ? 1 : side);`

What do you suggest?

Reply:
(To be clear, I don't like the hack I proposed but I'm not sure about any solution that would not be too cost-heavy)

Reply:
Done what you suggested! Feel free to approve if the PR is now okay with you